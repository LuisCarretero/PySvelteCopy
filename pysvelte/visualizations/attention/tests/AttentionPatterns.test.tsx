import { setBackend, ones } from "@tensorflow/tfjs";
import { render, screen } from "@testing-library/react";
import React from "react";
import { AttentionPatterns, colorAttentionTensors } from "../AttentionPatterns";

beforeAll(() => {
  // Use the node backend whilst testing
  setBackend("cpu");
});

describe("colorAttentionTensors", () => {
  it("creates a tensor of the correct shape", () => {
    // Input with 2 tokens, 2 heads
    const input = [
      [
        [1, 0],
        [0.5, 0.5]
      ],
      [
        [1, 0],
        [0.5, 0.5]
      ]
    ];

    const res = colorAttentionTensors(input);
    expect(res.shape).toEqual([2, 2, 2, 3]);
  });
});

describe("AttentionPatterns", () => {
  it("renders", () => {
    const tokens = ["A", "B", "C", "D"];
    const attention = ones([4, 4, 16]);

    render(
      <AttentionPatterns
        tokens={JSON.stringify(tokens)}
        attention={JSON.stringify(attention.arraySync())}
      />
    );

    // Check the header text loads
    screen.getByText("Attention Patterns");
  });
});
