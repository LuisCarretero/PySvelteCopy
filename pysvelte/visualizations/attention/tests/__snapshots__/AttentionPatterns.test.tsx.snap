// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colorAttentionTensors creates colors for heads matching the snapshot: 
      [
        [
          [
            [
              0,
              0,
              0,
            ],
            [
              255,
              0,
              0,
            ],
          ],
          [
            [
              0,
              0,
              0,
            ],
            [
              255,
              0,
              0,
            ],
          ],
        ],
        [
          [
            [
              255,
              255,
              255,
            ],
            [
              0,
              255,
              255,
            ],
          ],
          [
            [
              255,
              255,
              255,
            ],
            [
              0,
              255,
              255,
            ],
          ],
        ],
      ]
     1`] = `
[
  [
    [
      [
        0,
        0,
        0,
      ],
      [
        230,
        25,
        25,
      ],
    ],
    [
      [
        0,
        0,
        0,
      ],
      [
        230,
        25,
        25,
      ],
    ],
  ],
  [
    [
      [
        255,
        255,
        255,
      ],
      [
        25,
        229,
        230,
      ],
    ],
    [
      [
        255,
        255,
        255,
      ],
      [
        25,
        229,
        230,
      ],
    ],
  ],
]
`;
